### ISSUE TITLE
Support new OSs - macOS Sonoma 14.6 - Update allocator module

### ISSUE THREAD

**Issue by jotacarma90** [wazuh/wazuh#25068]
# Description
| Related issue |
|---|
| https://github.com/wazuh/wazuh/issues/25052 |

| Agent tier | Central components support | OS type |
|-|-|-|
| 1 | No | Minor |

# Plan

**DevOps**
- [x] Update the allocator images.
- [ ] ~~Update AMI, OVA, or Docker images if needed.~~

**Comment by davidcr01** [comment]
# Update Report

The box was created from a macOS 14.6 VM (upgraded from macOS 14). Then, I inserted the host key with the `ssh-copy-id -i /private/var/root/.ssh/vagrant_rsa vagrant@<VM_IP_ADDRESS>` and I created the box with the `tar` command.

## macStadium ARM

```
sh-3.2# vagrant --name="macos-test-1460" up macos-1460
Bringing machine 'macos-1460' up with 'parallels' provider...
==> macos-1460: Registering VM image from the base box 'macos-1460'...
==> macos-1460: Creating new virtual machine as a full clone of the box image...
==> macos-1460: Unregistering the box VM image...
==> macos-1460: Setting the default configuration for VM...
==> macos-1460: Setting the name of the VM: macos-test-1460
==> macos-1460: Preparing network interfaces based on configuration...
    macos-1460: Adapter 0: shared
    macos-1460: Adapter 1: hostonly
    macos-1460: Adapter 2: bridged
==> macos-1460: Clearing any previously set network interfaces...
==> macos-1460: Running 'pre-boot' VM customizations...
==> macos-1460: Booting VM...
==> macos-1460: Waiting for machine to boot. This may take a few minutes...
    macos-1460: SSH address: 192.168.64.5:22
    macos-1460: SSH username: vagrant
    macos-1460: SSH auth method: private key
==> macos-1460: Machine booted and ready!
==> macos-1460: Checking for Parallels Tools installed on the VM...
==> macos-1460: Setting hostname...
==> macos-1460: Configuring and enabling network interfaces...
sh-3.2# vagrant ssh macos-1460
Last login: Thu Aug  8 04:08:33 2024
vagrant@macos-1460 ~ % sw_vers
ProductName:            macOS
ProductVersion:         14.6.1
BuildVersion:           23G93
```

- The box is placed in the `/Users/jenkins/vagrant_boxes/macos1460.box`
- The box is imported with the `macos-1460` name.
- The box is NOT uploaded to S3 until we have a final release for this OS.

The machine was tested using the allocator module:
```
(wazuh-qa) ➜  deployability git:(maintenance/ww-25068-add-sonoma-14.6-allocator) ✗ python3 modules/allocation/main.py --action create --provider 'vagrant' --size 'large' --composite-name 'macos-sonoma-14.6-arm64' --label-termination-date '1d' --label-team  'devops' --label-issue 'https://github.com/wazuh/wazuh/issues/25068' 
/home/davidcr01/Wazuh/wazuh-qa/lib/python3.10/site-packages/paramiko/pkey.py:100: CryptographyDeprecationWarning: TripleDES has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.TripleDES and will be removed from this module in 48.0.0.
  "cipher": algorithms.TripleDES,
/home/davidcr01/Wazuh/wazuh-qa/lib/python3.10/site-packages/paramiko/transport.py:259: CryptographyDeprecationWarning: TripleDES has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.TripleDES and will be removed from this module in 48.0.0.
  "class": algorithms.TripleDES,
[2024-08-09 12:47:18] [DEBUG] SPNEGO._GSS: Python gssapi not available, cannot use any GSSAPIProxy protocols: No module named 'gssapi'
[2024-08-09 12:47:18] [DEBUG] SPNEGO._GSS: Python gssapi IOV extension not available: No module named 'gssapi'
[2024-08-09 12:47:18] [INFO] ALLOCATOR: Creating instance at /tmp/wazuh-qa
[2024-08-09 12:47:25] [INFO] ALLOCATOR: macStadium ARM server has less than 2 VMs running, deploying in this host.
[2024-08-09 12:47:25] [DEBUG] ALLOCATOR: Checking if instance directory exists on remote host
[2024-08-09 12:47:27] [DEBUG] ALLOCATOR: Creating instance directory on remote host
[2024-08-09 12:47:30] [DEBUG] ALLOCATOR: No config provided. Generating from payload
[2024-08-09 12:47:30] [DEBUG] ALLOCATOR: Generating new key pair
[2024-08-09 12:47:33] [DEBUG] ALLOCATOR: Vagrantfile created. Creating instance.
[2024-08-09 12:47:43] [INFO] ALLOCATOR: Instance wazuh-25068-sonoma-14.6-6284 created.
[2024-08-09 12:49:23] [INFO] ALLOCATOR: Instance wazuh-25068-sonoma-14.6-6284 started.
[2024-08-09 12:49:37] [INFO] ALLOCATOR: The inventory file generated at /tmp/wazuh-qa/wazuh-25068-sonoma-14.6-6284/inventory.yaml
[2024-08-09 12:49:37] [INFO] ALLOCATOR: The track file generated at /tmp/wazuh-qa/wazuh-25068-sonoma-14.6-6284/track.yaml
[2024-08-09 12:49:40] [INFO] ALLOCATOR: SSH connection successful.
[2024-08-09 12:49:40] [INFO] ALLOCATOR: Instance wazuh-25068-sonoma-14.6-6284 created successfully.

(wazuh-qa) ➜  deployability git:(maintenance/ww-25068-add-sonoma-14.6-allocator) ✗ ssh vagrant@10.10.0.250 -p43223
(vagrant@10.10.0.250) Password:
Last login: Thu Aug  8 04:09:25 2024
vagrant@macos-14 ~ % sw_vers
ProductName:            macOS
ProductVersion:         14.6.1
BuildVersion:           23G93
vagrant@macos-14 ~ % 
```

**Comment by davidcr01** [comment]
# Update Report

## macStadium Intel :green_circle:

After re-building the Vagrant box (now the VM was created from a Ventura machine), the problem was fixed:

```
jenkins@48503 parallels-testing % sudo vagrant --name="macos-sonoma-ww-25068" --bind-port='21795' --bind-ip='10.10.0.249' up macos-1460       
Bringing machine 'macos-1460' up with 'parallels' provider...
==> macos-1460: Registering VM image from the base box 'development/macos-sonoma-1460'...
==> macos-1460: Creating new virtual machine as a full clone of the box image...
==> macos-1460: Unregistering the box VM image...
==> macos-1460: Setting the default configuration for VM...
==> macos-1460: Setting the name of the VM: macos-sonoma-ww-25068
==> macos-1460: Fixed port collision for 22 => 2222. Now on port 2202.
==> macos-1460: Preparing network interfaces based on configuration...
    macos-1460: Adapter 0: shared
==> macos-1460: Clearing any previously set network interfaces...
==> macos-1460: Forwarding ports...
    macos-1460: 22 => 21795
==> macos-1460: Running 'pre-boot' VM customizations...
==> macos-1460: Booting VM...
==> macos-1460: Waiting for machine to boot. This may take a few minutes...
    macos-1460: SSH address: 10.211.55.148:22
    macos-1460: SSH username: vagrant
    macos-1460: SSH auth method: private key
==> macos-1460: Machine booted and ready!
==> macos-1460: Checking for Parallels Tools installed on the VM...
==> macos-1460: Setting hostname...
jenkins@48503 parallels-testing % sudo vagrant ssh macos-1460
Last login: Mon Aug 12 05:44:09 2024 from 10.211.55.2
vagrant@macos-1460 ~ % sw_vers
ProductName:            macOS
ProductVersion:         14.6.1
BuildVersion:           23G93
vagrant@macos-1460 ~ % 
```

```
(wazuh-qa) ➜  deployability git:(maintenance/ww-25068-add-sonoma-14.6-allocator) ✗ python3 modules/allocation/main.py --action create --provider 'vagrant' --size 'large' --composite-name 'macos-sonoma-14.6-amd64' --label-termination-date '1d' --label-team  'devops' --label-issue 'https://github.com/wazuh/wazuh/issues/25068'
/home/davidcr01/Wazuh/wazuh-qa/lib/python3.10/site-packages/paramiko/pkey.py:100: CryptographyDeprecationWarning: TripleDES has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.TripleDES and will be removed from this module in 48.0.0.
  "cipher": algorithms.TripleDES,
/home/davidcr01/Wazuh/wazuh-qa/lib/python3.10/site-packages/paramiko/transport.py:259: CryptographyDeprecationWarning: TripleDES has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.TripleDES and will be removed from this module in 48.0.0.
  "class": algorithms.TripleDES,
[2024-08-12 16:06:00] [DEBUG] SPNEGO._GSS: Python gssapi not available, cannot use any GSSAPIProxy protocols: No module named 'gssapi'
[2024-08-12 16:06:00] [DEBUG] SPNEGO._GSS: Python gssapi IOV extension not available: No module named 'gssapi'
[2024-08-12 16:06:00] [INFO] ALLOCATOR: Creating instance at /tmp/wazuh-qa
[2024-08-12 16:06:11] [INFO] ALLOCATOR: Using the macStadium Intel server to deploy.
[2024-08-12 16:06:11] [DEBUG] ALLOCATOR: Checking if instance directory exists on remote host
[2024-08-12 16:06:14] [DEBUG] ALLOCATOR: Creating instance directory on remote host
[2024-08-12 16:06:17] [DEBUG] ALLOCATOR: No config provided. Generating from payload
[2024-08-12 16:06:17] [DEBUG] ALLOCATOR: Generating new key pair
[2024-08-12 16:06:20] [DEBUG] ALLOCATOR: Vagrantfile created. Creating instance.
[2024-08-12 16:06:31] [INFO] ALLOCATOR: Instance wazuh-25068-sonoma-14.6-4577 created.
[2024-08-12 16:07:14] [INFO] ALLOCATOR: Instance wazuh-25068-sonoma-14.6-4577 started.
[2024-08-12 16:07:20] [DEBUG] ALLOCATOR: Instance wazuh-25068-sonoma-14.6-4577 is not running.                                Starting it.
[2024-08-12 16:08:19] [INFO] ALLOCATOR: The inventory file generated at /tmp/wazuh-qa/wazuh-25068-sonoma-14.6-4577/inventory.yaml
[2024-08-12 16:08:19] [INFO] ALLOCATOR: The track file generated at /tmp/wazuh-qa/wazuh-25068-sonoma-14.6-4577/track.yaml
[2024-08-12 16:08:22] [INFO] ALLOCATOR: SSH connection successful.
[2024-08-12 16:08:22] [INFO] ALLOCATOR: Instance wazuh-25068-sonoma-14.6-4577 created successfully.

(wazuh-qa) ➜  deployability git:(maintenance/ww-25068-add-sonoma-14.6-allocator) ✗ ssh vagrant@10.10.0.249 -p43220     
(vagrant@10.10.0.249) Password:
Last login: Mon Aug 12 05:44:09 2024 from 10.211.55.2
vagrant@macos-1460 ~ % sw_vers
ProductName:            macOS
ProductVersion:         14.6.1
BuildVersion:           23G93
vagrant@macos-1460 ~ % 
```

- The box is placed in the /Users/jenkins/vagrant_boxes/macos_sonoma_1460.box
- The box is imported with the `development/macos-sonoma-1460` name.
- The box is NOT uploaded to S3 until we have a final release for this OS.

**Comment by teddytpc1** [comment]
A new macOS Sonoma (14.6) AMI is available: `ami-0c82be92b15ae585f`.

**Comment by davidcr01** [comment]
The macOS AMIs were updated here: https://github.com/wazuh/wazuh-qa/pull/5671/commits/8ca26adf7ff4a2e27256dc3f857a135501305faa.
